packages/core/lib/command-utils.js

```diff
@@ -201,6 +203,41 @@ const runCommand = async function (command, options) {
   return await command.run(options);
 };
 
+const processHelpInput = inputStrings => {
+  //is it wrong syntax of help?
+  if (inputStrings.length > 2 && ["help", "--help"].includes(inputStrings[1])) {
+    const help = require("./commands/help").meta;
+    console.log(
+      "Error: please use below syntax to displaying help information\n",
+      "\n               ",
+      help.help.usage
+    );
+    process.exit();
+  }
+
+  // when `truffle --help <cmd>` is used, convert inputStrings to run as `truffle help <cmd>`
+  if (inputStrings.length > 1 && inputStrings[0] === "--help") {
+    inputStrings[inputStrings.indexOf("--help")] = "help";
+  }
+
+  // when `truffle <cmd> help | --help` is used, convert inputStrings  to run as `truffle help <cmd>`
+  if (["help", "--help"].includes(inputStrings[inputStrings.length - 1])) {
+    inputStrings.pop();
+    inputStrings.unshift("help");
+  }
+
+  const userWantsGeneralHelp =
+    inputStrings.length === 0 ||
```

Comment:
this condition will never be satisfied since we check to see whether "help" or "--help" was entered in order to determine whether to call `processHelpInput` in the first place

Reply:
nice catch. Simplifying it now.